services/worker_service.rb

```diff
@@ -8,44 +9,61 @@ module FastlaneCI
+    def start_workers_for_project_and_credential(project: nil, provider_credential: nil)
       user_responsible = provider_credential.ci_user
-      worker_key = resource_workers_dictionary_key(project: project, user_responsible: user_responsible)
-      raise "Worker already exists for project: #{project.name}, for user #{user_responsible.email}" unless self.resource_workers_dictionary[worker_key].nil?
+      workers_key = project_to_workers_dictionary_key(project: project, user_responsible: user_responsible)
+      raise "Worker already exists for project: #{project.name}, for user #{user_responsible.email}" unless self.project_to_workers_dictionary[workers_key].nil?
 
       repo_config = project.repo_config
       raise "incompatible repo_config and provider_credential" if provider_credential.type != repo_config.provider_credential_type_needed
 
       logger.debug("Starting worker for #{project.project_name}, on behalf of #{user_responsible.email}")
-      new_worker = nil
+      new_workers = []
       case provider_credential.type
       when FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
-        new_worker = FastlaneCI::CheckForNewCommitsOnGithubWorker.new(provider_credential: provider_credential, project: project)
+        if self.project_has_trigger_type(project: project, trigger_type: FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit])
+          new_workers << FastlaneCI::CheckForNewCommitsOnGithubWorker.new(provider_credential: provider_credential, project: project)
+        end
+        if self.project_has_trigger_type(project: project, trigger_type: FastlaneCI::JobTrigger::TRIGGER_TYPE[:nightly])
+          new_workers << FastlaneCI::NightlyBuildGithubWorker.new(provider_credential: provider_credential, project: project)
+        end
       else
         raise "unrecognized provider_type: #{provider_credential.type}"
       end
 
-      self.resource_workers_dictionary[worker_key] = new_worker
+      self.project_to_workers_dictionary[workers_key] = new_workers
     end
 
-    def stop_worker(project: nil, user_responsible: nil)
-      worker_key = resource_workers_dictionary_key(project: project, user_responsible: user_responsible)
-      worker = self.resource_workers_dictionary[worker_key]
+    def stop_workers(project: nil, user_responsible: nil)
+      workers_key = project_to_workers_dictionary_key(project: project, user_responsible: user_responsible)
+      workers = self.project_to_workers_dictionary[workers_key]
 
       # worker will die, may take up to `timeout` seconds
-      worker.die!
+      workers.each(&:die!)
 
-      self.resource_workers_dictionary[worker_key] = nil
+      self.project_to_workers_dictionary[workers_key] = nil
+    end
+
+    def num_workers
+      self.project_to_workers_dictionary.values.reduce(0) do |sum, workers|
+        sum + workers.length
+      end
+    end
+
+    def project_has_trigger_type(project: nil, trigger_type: nil)
```

Comment:
Methods that return booleans should end in `?`

Reply:
Nice catch 👍

Reply:
fixed.